dm/master/server.go

```diff
@@ -1312,6 +1317,8 @@ func (s *Server) resolveDDLLock(ctx context.Context, lockID string, replaceOwner
 		return nil, errors.Errorf("worker %s not waiting for DDL lock %s", owner, lockID)
 	}
 
+	traceGID := s.idGen.NextID("resolveDDLLock", 0)
```

Comment:
1. if call `resolveDDLLock` failed the first time, then call again, it will generate a new ID? and whether having side effect? if no side effect, maybe add a comment?

2. generate until it needed? like before L1343?

Reply:
It is a little complicated for `resolveDDLLock` failed, it may fail in first call, or some are success, but others fail, which leads to brain split of tracing events. 
I leave a comment here, I will try to find a better way to combine all distributed tracing events into a logic tracing group.